includes/login-register.php

```diff
@@ -64,63 +84,68 @@ function edd_register_form( $redirect = '' ) {
  * Process Login Form
  *
  * @since 1.0
+ * @since 2.9.24 No longer does validation which would prevent bruteforce detection plugins to be able to integrate.
+ *
  * @param array $data Data sent from the login form
  * @return void
 */
 function edd_process_login_form( $data ) {
-	if ( wp_verify_nonce( $data['edd_login_nonce'], 'edd-login-nonce' ) ) {
-		$user_data = get_user_by( 'login', $data['edd_user_login'] );
-		if ( ! $user_data ) {
-			$user_data = get_user_by( 'email', $data['edd_user_login'] );
-		}
-		if ( $user_data ) {
-			$user_ID = $user_data->ID;
-			$user_email = $user_data->user_email;
-
-			if ( wp_check_password( $data['edd_user_pass'], $user_data->user_pass, $user_data->ID ) ) {
-
-				if ( isset( $data['rememberme'] ) ) {
-					$data['rememberme'] = true;
-				} else {
-					$data['rememberme'] = false;
-				}
-
-				edd_log_user_in( $user_data->ID, $data['edd_user_login'], $data['edd_user_pass'], $data['rememberme'] );
-			} else {
-				edd_set_error( 'password_incorrect', __( 'The password you entered is incorrect', 'easy-digital-downloads' ) );
-			}
-		} else {
-			edd_set_error( 'username_incorrect', __( 'The username you entered does not exist', 'easy-digital-downloads' ) );
-		}
+
+	if ( ! empty( $data['edd_login_nonce'] ) && wp_verify_nonce( $data['edd_login_nonce'], 'edd-login-nonce' ) ) {
+		$rememberme = isset( $data['rememberme'] );
+
+		$user = edd_log_user_in( 0, $data['edd_user_login'], $data['edd_user_pass'], $rememberme );
+
 		// Check for errors and redirect if none present
 		$errors = edd_get_errors();
 		if ( ! $errors ) {
-			$redirect = apply_filters( 'edd_login_redirect', $data['edd_redirect'], $user_ID );
-			wp_redirect( $redirect );
-			edd_die();
+				$redirect = apply_filters( 'edd_login_redirect', $data['edd_redirect'], 0 );
+				wp_redirect( $redirect );
+				edd_die();
```

Comment:
phpcs says, "It is also important to remember to call exit() after a redirect so that no other unwanted code is executed." ... is calling `wp/edd_die` equivalent?

Reply:
@robincornett Yes
```
/**
 * Wrapper function for wp_die(). This function adds filters for wp_die() which
 * kills execution of the script using wp_die(). This allows us to then to work
 * with functions using edd_die() in the unit tests.
 *
 * @author Sunny Ratilal
 * @since 1.6
 * @return void
 */
function edd_die( $message = '', $title = '', $status = 400 ) {
	add_filter( 'wp_die_ajax_handler', '_edd_die_handler', 10, 3 );
	add_filter( 'wp_die_handler'     , '_edd_die_handler', 10, 3 );
	add_filter( 'wp_die_json_handler', '_edd_die_handler', 10, 3 );

	wp_die( $message, $title, array( 'response' => $status ));
}

```

Reply:
Thanks--I was not sure if `die` and `exit` were equivalent, but I see in the PHP manual that they are.